yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -314,634 +329,565 @@ describe('AVM simulator: transpiled Noir contracts', () => {
+        );
+      };
 
-      // TODO: change this once we don't force rethrowing of exceptions.
-      // Outer frame should not revert, but inner should, so the forwarded return value is 0
-      // expect(results.revertReason).toBeUndefined();
-      // expect(results.reverted).toBe(false);
-      expect(results.reverted).toBe(true);
-      expect(results.revertReason?.message).toEqual('Not enough L2GAS gas left');
-    });
+      it(`Nested call`, async () => {
+        const calldata = [value0, value1];
+        const context = createContext(calldata);
+        const callBytecode = getAvmTestContractBytecode('nested_call_to_add');
+        const addBytecode = getAvmTestContractBytecode('add_args_return');
+        mockGetBytecode(hostStorage, addBytecode);
+        const nestedTrace = mock<PublicSideEffectTraceInterface>();
+        mockTraceFork(trace, nestedTrace);
 
-    it(`Nested call`, async () => {
-      const calldata: Fr[] = [new Fr(1), new Fr(2)];
-      const callBytecode = getAvmTestContractBytecode('nested_call_to_add');
-      const addBytecode = getAvmTestContractBytecode('add_args_return');
-      const context = initContext({ env: initExecutionEnvironment({ calldata }) });
-      jest
-        .spyOn(context.persistableState.hostStorage.contractsDb, 'getBytecode')
-        .mockReturnValue(Promise.resolve(addBytecode));
+        const results = await new AvmSimulator(context).executeBytecode(callBytecode);
+        expect(results.reverted).toBe(false);
+        expect(results.output).toEqual([sum]);
 
-      const results = await new AvmSimulator(context).executeBytecode(callBytecode);
+        expectTracedNestedCall(context.environment, nestedTrace);
+      });
 
-      expect(results.reverted).toBe(false);
-      expect(results.output).toEqual([new Fr(3)]);
-    });
+      it(`Nested static call`, async () => {
+        const calldata = [value0, value1];
+        const context = createContext(calldata);
+        const callBytecode = getAvmTestContractBytecode('nested_static_call_to_add');
+        const addBytecode = getAvmTestContractBytecode('add_args_return');
+        mockGetBytecode(hostStorage, addBytecode);
+        const nestedTrace = mock<PublicSideEffectTraceInterface>();
+        mockTraceFork(trace, nestedTrace);
 
-    it(`Nested static call`, async () => {
-      const calldata: Fr[] = [new Fr(1), new Fr(2)];
-      const callBytecode = getAvmTestContractBytecode('nested_static_call_to_add');
-      const addBytecode = getAvmTestContractBytecode('add_args_return');
-      const context = initContext({ env: initExecutionEnvironment({ calldata }) });
-      jest
-        .spyOn(context.persistableState.hostStorage.contractsDb, 'getBytecode')
-        .mockReturnValue(Promise.resolve(addBytecode));
+        const results = await new AvmSimulator(context).executeBytecode(callBytecode);
+        expect(results.reverted).toBe(false);
+        expect(results.output).toEqual([sum]);
```

Comment:
For these [sum] cases, might be worth just doing [value0.add(value1)] and removing the sum const.